README.md
# bee-affix

[![npm version](https://img.shields.io/npm/v/bee-affix.svg)](https://www.npmjs.com/package/bee-affix)
[![Build Status](https://img.shields.io/travis/tinper-bee/bee-affix/master.svg)](https://travis-ci.org/tinper-bee/bee-affix)
[![Coverage Status](https://coveralls.io/repos/github/tinper-bee/bee-affix/badge.svg?branch=master)](https://coveralls.io/github/tinper-bee/bee-affix?branch=master)
[![devDependency Status](https://img.shields.io/david/dev/tinper-bee/bee-affix.svg)](https://david-dm.org/tinper-bee/bee-affix#info=devDependencies)
[![NPM downloads](http://img.shields.io/npm/dm/bee-affix.svg?style=flat)](https://npmjs.org/package/bee-affix)
[![Average time to resolve an issue](http://isitmaintained.com/badge/resolution/tinper-bee/bee-affix.svg)](http://isitmaintained.com/project/tinper-bee/bee-affix "Average time to resolve an issue")
[![Percentage of issues still open](http://isitmaintained.com/badge/open/tinper-bee/bee-affix.svg)](http://isitmaintained.com/project/tinper-bee/bee-affix "Percentage of issues still open")


react bee-affix component for tinper-bee

some description...

## Affix
将页面元素钉在可视范围。
当内容区域比较长，需要滚动页面时，这部分内容对应的操作或者导航需要在滚动范围内始终展现。常用于侧边菜单和按钮组合。页面可视范围过小时，慎用此功能以免遮挡页面内容。

## 使用方法

```
import Affix from 'bee-affix';

React.render(<Affix><div className="content"></div></Affix>,
         document.getElementById('target'));
```

## 样式引入

```
<link rel="stylesheet" href="./node_modules/bee-affix/build/Affix.css">
//或是
import "./node_modules/bee-affix/src/Affix.scss"
//或是
import "./node_modules/bee-affix/build/Affix.css"
```

## API

|参数|说明|类型|默认值|
|:--|:---:|:--:|---:|
|container|固定元素所属容器|object|document.body|
|offsetTop|固定被触发距离，默认是0 ，可选的属性|number|0|
|horizontal|被固定时是否可以左右滑动，默认是false，可选的属性|boolean|false|
|canHidden|当Affix的内容高度超过container可视区域的高度，Affix是否可以被隐藏一部分内容,false：affix始终完全展示，true：展示部分受container限制|boolean|false|
|target|设置 Affix 需要监听其滚动事件的元素，值为一个返回对应 DOM 元素的函数，默认值() => window |function|() => window|
|onChange|触发固定或者固定取消时的回调函数,返回值{ affixed: true, event: evt} |function|-|
|onTargetChange|时刻获取Affix的状态值,返回值{state} |function|-|
#### 开发调试

```sh
$ npm install -g bee-tools
$ git clone https://github.com/tinper-bee/bee-affix
$ cd bee-affix
$ npm install
$ npm run dev
```


CHANGELOG.md
## [1.0.20](https://github.com/tinper-bee/bee-affix/compare/v1.0.19...v1.0.20) (2021-04-30)


### Bug Fixes

* 编译代码 ([70bbe92](https://github.com/tinper-bee/bee-affix/commit/70bbe92bb9348daf6739c2b5be20b04bed10d6c1))



## [1.0.19](https://github.com/tinper-bee/bee-affix/compare/v1.0.18...v1.0.19) (2021-04-30)


### Bug Fixes

* 处理对象变化判断条件 ([73cd7bb](https://github.com/tinper-bee/bee-affix/commit/73cd7bbc3183d3ec96a7f0209e077f8dfb538f57))



## [1.0.18](https://github.com/tinper-bee/bee-affix/compare/v1.0.17...v1.0.18) (2020-04-24)


### Bug Fixes

* 自动化 ([ec7c187](https://github.com/tinper-bee/bee-affix/commit/ec7c1874320cc69384d510f3a55432f135deba23))



<a name="1.0.17"></a>
## [1.0.17](https://github.com/tinper-bee/bee-affix/compare/v1.0.16...v1.0.17) (2019-09-23)



<a name="1.0.16"></a>
## [1.0.16](https://github.com/tinper-bee/bee-affix/compare/1.0.15...v1.0.16) (2019-09-11)



<a name="1.0.15"></a>
## [1.0.15](https://github.com/tinper-bee/bee-affix/compare/v1.0.14...1.0.15) (2019-03-14)



<a name="1.0.14"></a>
## [1.0.14](https://github.com/tinper-bee/bee-affix/compare/v1.0.12...v1.0.14) (2018-11-16)



<a name="1.0.12"></a>
## [1.0.12](https://github.com/tinper-bee/bee-affix/compare/v1.0.11...v1.0.12) (2018-10-10)



<a name="1.0.11"></a>
## [1.0.11](https://github.com/tinper-bee/bee-affix/compare/v1.0.10...v1.0.11) (2018-10-10)



<a name="1.0.10"></a>
## [1.0.10](https://github.com/tinper-bee/bee-affix/compare/v1.0.5...v1.0.10) (2018-10-10)



<a name="1.0.5"></a>
## [1.0.5](https://github.com/tinper-bee/bee-affix/compare/v1.0.4...v1.0.5) (2018-08-18)



<a name="1.0.4"></a>
## [1.0.4](https://github.com/tinper-bee/bee-affix/compare/v1.0.3...v1.0.4) (2018-08-14)



<a name="1.0.3"></a>
## 1.0.3 (2018-08-10)





src/Affix.js
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import {addEventListener,removeEventListener} from './utils';
const propTypes = {
    container: PropTypes.object,
    offsetTop: PropTypes.number,
    horizontal: PropTypes.bool,
    target: PropTypes.func,//不对外开放，获取滚动scroll以及resize功能
    onChange: PropTypes.func,//状态fixed或infixed时候调用
    onTargetChange: PropTypes.func,//功能只有一个，时时刻刻输出state的状态
    zIndex: PropTypes.number,
    canHidden:PropTypes.bool,
    childrenRef:PropTypes.object,
};

const defaultProps = {
    offsetTop: 0,
    horizontal: false,
    // container: document.body,
    target: () => window,
    onChange: (affixed) => ({}),
    onTargetChange: (state) => ({}),
    zIndex: 2,
    canHidden:false,
    childrenRef:null,
};

class Affix extends Component {
	constructor(props) {
        super(props);
        this.state = {
            affixed: false,
            initTop: 0,
            initLeft: 0,
            top:0,//affix距离顶部的距离
            left:0,//affix距离左边的距离
            marginTop: 0,//top - containerTop
            marginLeft: 0,//left - containerLeft
            height: 0,//affix的高度
            width: 0,//affix的宽度
            containerHeight: 0,//container的高度
            containerWidth: 0,//container的宽度
            containerId:"u-affix-container"+Math.random().toString(26).substring(2,10),
        }
        this.calculate = this.calculate.bind(this);
        this.getInitPosition = this.getInitPosition.bind(this);
        this.getContainerDOM = this.getContainerDOM.bind(this);
        this.handleTargetChange = this.handleTargetChange.bind(this);
    }
 

    componentDidMount() {
        this.getInitPosition(undefined,true);
        const listenTarget = this.props.target();
        if (listenTarget) {
            addEventListener(listenTarget,'resize',this.handleTargetChange);
            addEventListener(listenTarget,'scroll',this.handleTargetChange);
            // listenTarget.addEventListener('resize', e=>this.handleTargetChange(e,'resize'))
            // listenTarget.addEventListener('scroll', e=>this.handleTargetChange(e,'scroll'))
        }
    }

    componentWillUnmount() {
        const listenTarget = this.props.target();
        if (listenTarget) {
            removeEventListener(listenTarget,'resize',this.handleTargetChange);
            removeEventListener(listenTarget,'scroll',this.handleTargetChange);
            // listenTarget.addEventListener('resize', e=>this.handleTargetChange(e,'resize'))
            // listenTarget.addEventListener('scroll', e=>this.handleTargetChange(e,'scroll'))
        }
    }

    /**
     * 获取container
     * @return {[type]} [description]
     */
    getContainerDOM =(nextProps)=> {
        const container = nextProps?nextProps.container:this.props.container;
        if(!container){
            return document.body
        }
        if (container != document.body) {
            return ReactDOM.findDOMNode(container);
        }
        return container;
    }

    /**
     * 第一次 和 每次srcoll和resize也会执行该方法
     * initTop,initLeft，marginTop,marginLeft都是不变的
     * @return {[type]} [description]
     */
    getInitPosition =(nextProps,flag)=> {
        const container = this.getContainerDOM(nextProps);
        // 20180927children是个变化，所以在nextprops要传入childrenRef，否则直接使用后面的语句
        const thisElm = nextProps && nextProps.childrenRef && ReactDOM.findDOMNode(nextProps.childrenRef) || ReactDOM.findDOMNode(this);
        this.setState({
            height: thisElm.offsetHeight,
            width: thisElm.offsetWidth,
            containerHeight: container.offsetHeight,
            containerWidth: container.offsetWidth,
        })
        const containerRect = container.getBoundingClientRect();
        const thisElemRect = thisElm.getBoundingClientRect();

        let { top, left } = thisElemRect;
        const marginTop = top - containerRect.top;
        const marginLeft = left - containerRect.left;
        this.setState({
            top: top,
            left: left,
            initTop: flag ? top :this.state.initTop,
            initLeft: flag ? left :this.state.initLeft,
            marginTop: marginTop,
            marginLeft: marginLeft
        });
    }

    /**
     * [description]主要用于处理scroll以及reseize事件重新计算布局
     * @param  {[object]} evt [scroll或者resize事件]
     * 有两个函数：onChange 和 onTargetChange;
     */
    handleTargetChange =(evt,type)=> {
        const container = this.getContainerDOM();//是body
        const { top, left } = container.getBoundingClientRect()

        this.setState({
            top: top + this.state.marginTop,//=0是临界值，滚动条使得屏幕顶端正好在affix上面，前者一直变化后者为不变
            left: left + this.state.marginLeft,//原理同上
            containerHeight: container.offsetHeight,
            containerWidth: container.offsetWidth,
        });

        if (this.state.top < this.props.offsetTop) {
            if ( this.state.affixed === false) {
                this.props.onChange({ affixed: true, event: evt});
            }
            this.setState({ affixed: true });//=true,calculate起作用
        }

        if (this.state.top >= this.props.offsetTop) {
            if ( this.state.affixed === true) {
                this.props.onChange({ affixed: false, event: evt});
                //新增还原样式
                
            }
            this.setState({ affixed: false })
        }
 
        this.props.onTargetChange(this.state);
        this.getInitPosition(undefined,false);
    }
    
    /**
     * 只有上面的方法handleTargetChange使得affixed=ture才会执行
     * @return {[type]} [description]
     */
    calculate =()=> {
        let fixStyle = {};
        let boxStyle = {};
        //20171102修改，添加(this.state.top - this.state.marginTop == 0)的判断，谨防height+offsetTop >= containerHeight, handleTargetChange中的
        //this.state.top <= this.props.offsetTop 恒成立，一直有position:affixed
        if(!this.state.affixed || (this.state.top - this.state.marginTop == 0)) return {fixStyle, boxStyle}
        let h = (this.state.top - this.state.marginTop + this.state.containerHeight) - this.state.height;
        if (this.state.top < this.props.offsetTop) {
            fixStyle = {
                position: "fixed",
                //修改20171102 去掉展示Affix全部内容，若是Affix内容高度大于container可展示，那么Affix只可展示部分
                // top: this.props.canHidden ? ( h < 0 ? h : Math.min(h, this.props.offsetTop)) : ( h < 0 ? 0 : Math.min(h, this.props.offsetTop)),
                top: this.props.canHidden ? ( h < 0 ? h : Math.min(h, this.props.offsetTop)) : this.props.offsetTop,
                left: this.props.horizontal ? this.state.initLeft : this.state.left,
                height: this.state.height,
                width: this.state.width,
                zIndex: this.props.zIndex,
            }
            boxStyle = {height: this.state.height}
        }
        return {fixStyle, boxStyle}
    }

    render() {
        const { fixStyle, boxStyle }= this.calculate();
        const {
            offsetTop,
            horizontal,
            container,
            target,
            onChange,
            onTargetChange,
            zIndex,
            canHidden,
            childrenRef,
            ...others
        } = this.props;
        return (
            <div {...others} className={classnames("u-affix-container",this.props.className)} id={this.state.containerId} style={boxStyle}>
                <div className="u-affix-content" style={fixStyle}>
                    {this.props.children}
                </div>
            </div>
        )
    }
}
Affix.propTypes = propTypes;

Affix.defaultProps = defaultProps;
export default Affix;

demo/demolist/Demo1.js
/**
*
* @title 基本的Affix
* @description 基本的Affix
*
*/

import React, { Component } from 'react';
import Affix from '../../src';


class Demo1 extends Component {
  constructor(props){
    super(props);
    this.state = {
      container:null,
    }
  }

  componentDidMount(){
    if(document.getElementById('outer-box')){
      this.setState({container:document.getElementById('outer-box')})
    }
  }
  render () {
  
    return (
      <div className="outer-box" id="outer-box">
        <label>某个div内的affix，container canHidden={true} zIndex={2001}</label>
          <Affix container={this.state.container} canHidden={true} zIndex={2001}>
            <div className='content'>
              <span>affix</span>
            </div>
          </Affix>
       
      </div>
    )
  }
}


export default Demo1;

demo/demolist/Demo2.js
/**
*
* @title offsetTop Affix
* @description 触发固定的top
*
*/

import React, { Component } from 'react';
import Affix from '../../src';

class Demo2 extends Component {
  render () {
    //请注意，这个container是为了适应tinperbee官网的布局特意设定，其他没有意外不需要传container，默认body
    let container = document.getElementsByClassName('page-container u-container example')[0]? document.getElementsByClassName('page-container u-container example')[0] : document.getElementById('tinperBeeDemo');

    return (
      <div className="outer-box" id="outer-box2">
        <label>基本的Affix，具体屏幕顶部距离等于offetTop `zIndex={2001} offsetTop=200`</label>
        <Affix container={container} zIndex={2001} offsetTop={200}>
          <div className='content'>
            <span>affix</span>
          </div>
        </Affix>
      </div>
    )
  }
}


export default Demo2;

demo/demolist/Demo3.js
/**
 *
 * @title horizontal Affix
 * @description 基本的Affix，水平滚动affix距离左侧位置确定
 *
 */


import React, { Component } from 'react';
import Affix from '../../src';

class Demo3 extends Component {
    render() {
       //请注意，这个container是为了适应tinperbee官网的布局特意设定，其他没有意外不需要传container，默认body
        let container = document.getElementsByClassName('page-container u-container example')[0]? document.getElementsByClassName('page-container u-container example')[0] : document.getElementById('tinperBeeDemo');

        return (
          <div className = "outer-box"id = "outer-box3" >
            <label > 基本的Affix，水平滚动affix距离左侧位置确定 `zIndex={2001} horizontal offsetTop=450 ` </label> 
            <div style={{width:"200px"}}>
              <Affix container={container} zIndex={2001} horizontal={true} offsetTop = { 450 } >
                <div className = 'content' >
                  <span > affix </span>
                </div> 
              </Affix> 
            </div>
          </div>
        )
    }
}

export default Demo3;

demo/demolist/Demo4.js
/**
*
* @title affix下面的dom是可变化的时候
* @description affix下面的dom是可变化的时候
*
*/

import React, { Component } from 'react';
import Button from 'bee-button';
import Affix from '../../src';


class Demo4 extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dom: null,
      childrenRef: null,
    }
  }

  componentDidMount() {
    if (document.getElementById('outer-box')) {
      this.setState({
        dom: <div style={{ backgroundColor: 'green', opacity: 0.2 }} ref={(ref) => { this.setState({ childrenRef: ref }) }} ><h1>affix</h1><h1>affix</h1></div>
      });
    }
    setTimeout(() => {
      this.setState({
        dom: <div style={{ backgroundColor: 'green', opacity: 0.2 }} ref={(ref) => { this.setState({ childrenRef: ref }) }} ><h1>affix</h1><h1>affix</h1><h1>affix</h1><h1>affix</h1><h1>affix</h1><h1>affix</h1></div>
      });
    }, 16000);

  }
  onChange = () => {
    console.log('hahah变化');
  }
  render() {
    return (
      <div className="outer-box" id="outer-box">
        <label>affix下面的dom是可变化的时候</label>
          <Affix zIndex={2001} onChange={this.onChange} childrenRef={this.state.childrenRef}>
              {this.state.dom}
          </Affix>

        <Button colors="primary">1222</Button>

      </div>
    );
  }

}


export default Demo4;

demo/index-demo-base.js

import { Con, Row, Col } from 'bee-layout';
import { Panel } from 'bee-panel';
import Button from 'bee-button';
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Affix from '../src';


const CARET = <i className="uf uf-arrow-down"></i>;

const CARETUP = <i className="uf uf-arrow-up"></i>;


{demolist}

class Demo extends Component {
    constructor(props){
        super(props);
        this.state = {
            open: false
        }
        this.handleClick = this.handleClick.bind(this);
    }
    handleClick() {
        this.setState({ open: !this.state.open })
    }

    render () {
        const { title, example, code, desc  } = this.props;
        let caret = this.state.open ? CARETUP : CARET;
        let text = this.state.open ? "隐藏代码" : "查看代码";

        const footer = (
            <Button shape="block" onClick={ this.handleClick }>
                { caret }
                { text }
            </Button>
        );
        return (
            <Col md={12} >
                <h3>{ title }</h3>
                <p>{ desc }</p>
                <Panel collapsible headerContent expanded={ this.state.open } colors='bordered' header={ example } footer={footer} footerStyle = {{padding: 0}}>
                    <pre><code className="hljs javascript">{ code }</code></pre>
                </Panel>
            </Col>
        )
    }
}

class DemoGroup extends Component {
    constructor(props){
        super(props)
    }
    render () {
        return (
                <Row>
                    {DemoArray.map((child,index) => {

                        return (
                            <Demo example= {child.example} title= {child.title} code= {child.code} desc= {child.desc} key= {index}/>
                        )

                    })}
                </Row>
        )
    }
}

ReactDOM.render(<DemoGroup/>, document.getElementById('tinperBeeDemo'));


demo/demolist/ProjectDemo1.js
/**
*
* @title 项目中的bugdemo
* @description 项目中的bugdemo
*
*/

import React, { Component } from 'react';
import Affix from '../../src';

class Demo1 extends Component {
  render () {
      let CONFIG = ['申请消息','其他消息','最好的消息']
      let chooseIndex = 1;
    return (
      <div className="outer-box" id="outer-box">
        <label>基本的Affix</label>
        <div className="vertical-tab">
        </div>
        <Affix>
          <h3 className="financeApp-title">
            <span className="financeApp-title-item title">贷款申请</span>
            <ul className="financeApp-tab cf">
                {CONFIG.map((item,index)=>{
                    return (<li key={index} className={index == chooseIndex?'active':''}>{item}</li>)
                })}
                <li className="scrollBar tabs-nav-animated" ref="navBar"></li>
            </ul>
            <div className="financeApp-title-item">
              dsdds
            </div>
          </h3>
        </Affix>
        <div className="content-ss">
          <div className="content-ss-item">
                  <label htmlFor="" className="name">手机号</label>
                  <input className="info"/>
            </div>
            <div className="content-ss-item">
                  <label htmlFor="" className="name">手机号2</label>
                  <input className="info"/>
            </div>
       </div>
      </div>
    )
  }
}


export default Demo1;

demo/index.js
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Con, Row, Col } from 'bee-layout';
import { Panel } from 'bee-panel';
import Drawer from 'bee-drawer';
import Clipboard from 'bee-clipboard'; 
import Button from '../src';



var Demo1 = require("./demolist/Demo1");var Demo2 = require("./demolist/Demo2");var Demo3 = require("./demolist/Demo3");var Demo4 = require("./demolist/Demo4");var DemoArray = [{"example":<Demo1 />,"title":" 基本的Affix,带有container","code":"/**\n*\n* @title 基本的Affix,带有container\n* @description 基本的Affix,带有container\n*\n*/\n\nimport React, { Component } from 'react';\nimport { Affix, Button } from 'tinper-bee';\n\n\nclass Demo1 extends Component {\n  constructor(props){\n    super(props);\n    this.state = {\n      container:null,\n    }\n  }\n\n  componentDidMount(){\n    if(document.getElementById('outer-box')){\n      this.setState({container:document.getElementById('outer-box')})\n    }\n  }\n  render () {\n  \n    return (\n      <div className=\"demo1\">\n        <div>某个div内的affix，container canHidden={true} zIndex={2001}</div>\n        <div className=\"outer-box checkered stripes\" id=\"outer-box\">\n          <Affix container={this.state.container} canHidden={true} zIndex={2001}>\n            <Button colors=\"primary\">affix in container</Button>\n          </Affix>\n        </div>\n      </div>\n    )\n  }\n}\n\n\n","desc":" 基本的Affix,带有container","scss_code":"// .content {\n//   width: 150px;\n//   height: 100px;\n//   background-image: linear-gradient(-180deg, #6873E8 0%, #277DF4 100%);\n//   font-size: 20px;\n//   color: #fff;\n//   text-align: center;\n//   line-height: 100px;\n// }\n// .u-panel-title{\n//   height: 800px;\n// }\n// #outer-box{\n//   width: 700px;\n//   height: 300px!important;\n//   background: #1baede;\n// }\n.stripes {   \n  width: 500px;   \n  height: 400px!important;;   \n  float: left;   \n     \n  margin: 10px;   \n     \n  -webkit-background-size: 16px 16px;   \n  -moz-background-size: 16px 16px;   \n  background-size: 16px 16px; /* 控制条纹的大小 */  \n     \n  -moz-box-shadow: 1px 1px 8px #ccc;   \n  -webkit-box-shadow: 1px 1px 8px #ccc;   \n  box-shadow: 1px 1px 8px #ccc;   \n}  \n.checkered {   \n  background-image: -webkit-gradient(linear, 0 0, 100% 100%, color-stop(.25, #ccc), color-stop(.25, transparent), to(transparent)),   \n                    -webkit-gradient(linear, 0 100%, 100% 0, color-stop(.25, #ccc), color-stop(.25, transparent), to(transparent)),   \n                    -webkit-gradient(linear, 0 0, 100% 100%, color-stop(.75, transparent), color-stop(.75, #ccc)),   \n                    -webkit-gradient(linear, 0 100%, 100% 0, color-stop(.75, transparent), color-stop(.75, #ccc));   \n  background-image: -moz-linear-gradient(45deg, #ccc 25%, transparent 25%, transparent),   \n                    -moz-linear-gradient(-45deg, #ccc 25%, transparent 25%, transparent),   \n                    -moz-linear-gradient(45deg, transparent 75%, #ccc 75%),   \n                    -moz-linear-gradient(-45deg, transparent 75%, #ccc 75%);   \n  background-image: -o-linear-gradient(45deg, #ccc 25%, transparent 25%, transparent),   \n                    -o-linear-gradient(-45deg, #ccc 25%, transparent 25%, transparent),   \n                    -o-linear-gradient(45deg, transparent 75%, #ccc 75%),   \n                    -o-linear-gradient(-45deg, transparent 75%, #ccc 75%);   \n  background-image: linear-gradient(45deg, #ccc 25%, transparent 25%, transparent),   \n                    linear-gradient(-45deg, #ccc 25%, transparent 25%, transparent),   \n                    linear-gradient(45deg, transparent 75%, #ccc 75%),   \n                    linear-gradient(-45deg, transparent 75%, #ccc 75%);   \n}  "},{"example":<Demo2 />,"title":" offsetTop Affix","code":"/**\n*\n* @title offsetTop Affix\n* @description 触发固定的距离屏幕顶部高度等于offetTop\n*\n*/\n\nimport React, { Component } from 'react';\nimport { Affix, Button } from 'tinper-bee';\n\nclass Demo2 extends Component {\n  render () {\n    //请注意，这个container是为了适应tinperbee官网的布局特意设定，其他没有意外不需要传container，默认body\n    let container = document.getElementsByClassName('page-container u-container example')[0]? document.getElementsByClassName('page-container u-container example')[0] : document.getElementById('tinperBeeDemo');\n\n    return (\n      <div className=\"demo2\" id=\"outer-box2\">\n        <label>基本的Affix，具体屏幕顶部距离等于offetTop `zIndex={2001} offsetTop=200`</label>\n        <Affix container={container} zIndex={2001} offsetTop={120}>\n          <Button colors=\"primary\">120px to affix top</Button>\n        </Affix>\n      </div>\n    )\n  }\n}\n\n\n","desc":" 触发固定的距离屏幕顶部高度等于offetTop","scss_code":"// .content {\n//     width: 150px;\n//     height: 100px;\n//     background-image: linear-gradient(-180deg, #6873E8 0%, #277DF4 100%);\n//     font-size: 20px;\n//     color: #fff;\n//     text-align: center;\n//     line-height: 100px;\n//   }"},{"example":<Demo3 />,"title":" horizontal Affix","code":"/**\n *\n * @title horizontal Affix\n * @description 基本的Affix，水平滚动affix距离左侧位置确定\n *\n */\n\n\nimport React, { Component } from 'react';\nimport { Affix, Button } from 'tinper-bee';\n\nclass Demo3 extends Component {\n    render() {\n       //请注意，这个container是为了适应tinperbee官网的布局特意设定，其他没有意外不需要传container，默认body\n        let container = document.getElementsByClassName('page-container u-container example')[0]? document.getElementsByClassName('page-container u-container example')[0] : document.getElementById('tinperBeeDemo');\n\n        return (\n          <div className = \"demo3\"id = \"outer-box3\" >\n            <label > 基本的Affix，水平滚动affix距离左侧位置确定 `zIndex={2001} horizontal offsetTop=450 ` </label> \n            <div className=\"affix-container\">\n              <Affix container={container} zIndex={2001} horizontal={true} offsetTop = { 450 } >\n                <Button colors=\"primary\">450px to affix top</Button>\n              </Affix> \n            </div>\n          </div>\n        )\n    }\n}\n\n","desc":" 基本的Affix，水平滚动affix距离左侧位置确定","scss_code":"// .content {\n//     width: 150px;\n//     height: 100px;\n//     background-image: linear-gradient(-180deg, #6873E8 0%, #277DF4 100%);\n//     font-size: 20px;\n//     color: #fff;\n//     text-align: center;\n//     line-height: 100px;\n//   }\n.demo3{\n  width: 2000px;\n  height: 400px;\n  .affix-container{\n    width: 200px;\n    margin-left: 100px;\n    margin-top: 300px;\n  }\n}"},{"example":<Demo4 />,"title":" 导航吸顶效果","code":"/**\n*\n* @title 导航吸顶效果\n* @description 结合Navbar使用示例\n*\n*/\n\nimport React, { Component } from 'react';\nimport { Affix, Button, Navbar } from 'tinper-bee';\n\nconst NavItem = Navbar.NavItem;\nconst Header = Navbar.Header;\nconst Brand = Navbar.Brand;\nconst Nav = Navbar.Nav;\n\nclass Demo4 extends Component {\n  constructor(props) {\n    super(props);\n    this.state = {\n      childrenRef: null,\n    }\n  }\n\n  onChange = () => {\n    console.log('触发固定或者固定取消时的回调函数');\n  }\n  render() {\n    return (\n      <div className=\"outer-box\" id=\"outer-box\">\n          <Affix zIndex={2001} onChange={this.onChange} childrenRef={this.state.childrenRef}>\n            <Navbar>\n              <Header>\n                <Brand>\n                    <a href=\"#\">React-FED</a>\n                </Brand>\n              </Header>\n              <Nav>\n                <NavItem eventKey={1}>选项1</NavItem>\n                <NavItem eventKey={2}>\n                    选项2\n                </NavItem>\n                <NavItem eventKey={3}>\n                    选项3\n                </NavItem>\n              </Nav>\n            </Navbar>\n          </Affix>\n        <p>问世间、情为何物，只教生死相许？天南地北双飞客，老翅几回寒暑。欢乐趣，离别苦， 就中更有痴儿女。君应有语，渺万里层云，千山暮雪，只影向谁去？横汾路，寂寞当年箫鼓，荒烟依旧平楚。招魂楚些何嗟及， 山鬼暗谛风雨。天也妒，未信与，莺几燕子俱黄土。千秋万古，为留待骚人，狂歌痛饮，来访雁丘处。</p>\n\n      </div>\n    );\n  }\n\n}\n\n\n","desc":" 结合Navbar使用示例","scss_code":"// body{\n//     height: 3000px;\n// }\n//   .content {\n//       width: 60%;\n//       background-image: linear-gradient(-180deg, #6873E8 0%, #277DF4 100%);\n//       font-size: 20px;\n//       color: #fff;\n//       text-align: center;\n//       /* line-height: 100px; */\n//     }\n//     .u-panel-title{\n//       height: 800px;\n//     }\n//     .button{\n//       height: 60px;\n//       width: 200px;\n//       background: pink;\n//       color:#fff;\n//       border-radius: 3px;\n//     }\n//     h1{\n//       margin:0;\n//       padding:0;\n//     }\n.navbar-container{\n    margin: 10px 15px;\n}"}]


class Demo extends Component {
    constructor(props){
        super(props);
        this.state = {
            open: false
        }
    }
    handleClick=()=> {
        this.setState({ open: !this.state.open })
    }
    fCloseDrawer=()=>{
        this.setState({
            open: false
        })
    }

    render () {
        const { title, example, code, desc, scss_code  } = this.props;

        const header = (
            <div>
                <p className='component-title'>{ title }</p>
                <p>{ desc }</p>
                <span className='component-code' onClick={this.handleClick}> 查看源码 <i className='uf uf-arrow-right'/> </span>
            </div>
        );
        return (
            <Col md={12} id={title.trim()} className='component-demo'>
            <Panel header={header}>
                {example}
            </Panel>
           
            <Drawer className='component-drawerc' title={title} show={this.state.open} placement='right' onClose={this.fCloseDrawer}>
            <div className='component-code-copy'> JS代码 
                <Clipboard action="copy" text={code}/>
            </div>
            <pre className="pre-js">
                <code className="hljs javascript">{ code }</code>
            </pre >
            {!!scss_code ?<div className='component-code-copy copy-css'> SCSS代码 
                <Clipboard action="copy" text={scss_code}/>
            </div>:null }
                { !!scss_code ? <pre className="pre-css">
                 <code className="hljs css">{ scss_code }</code>
                 </pre> : null }
            </Drawer>
        </Col>
    )
    }
}

class DemoGroup extends Component {
    constructor(props){
        super(props)
    }
    render () {
        return (
            <Row>
            {DemoArray.map((child,index) => {

                return (
            <Demo example= {child.example} title= {child.title} code= {child.code} scss_code= {child.scss_code} desc= {child.desc} key= {index}/>
    )

    })}
    </Row>
    )
    }
}

ReactDOM.render(<DemoGroup/>, document.getElementById('tinperBeeDemo'));
